### PR TITLE
Velodrome LP Compounder

### PR DESCRIPTION
**Pull request by amatureApe**

### CODE REVIEW

File: src/vault/strategy/compounder/velodrome/VelodromeCompounder.sol [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+        // Trade rewards for base asset
+        address[] memory rewardTokens = IWithRewards(address(this))
+            .rewardTokens();
+
+        uint256 len = rewardTokens.length;
+        for (uint256 i = 0; i < len; i++) {
+            uint256 rewardBal = IERC20(rewardTokens[i]).balanceOf(
+                address(this)
+            );
+            uint256 bal0 = rewardBal / 2;
```
Comment (RedVeil) [inline src/vault/strategy/compounder/velodrome/VelodromeCompounder.sol:182]: Why are we splitting this? I think we should just trade the full balance for the base asset. Only if we create an LP token after we need to seperate balances but that would be in `_getAsset` of the lp compounder

File: src/vault/strategy/compounder/velodrome/VelodromeLPCompounder.sol [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+contract VelodromeLpCompounder is VelodromeCompounder {
+    function _verifyAsset(
+        address baseAsset,
+        address asset,
+        bytes memory toAssetPath,
+        bytes memory optionalData
+    ) internal override {
+        address velodromeRouter = abi.decode(optionalData, (address));
+
+        if (IVelodromeRouter(velodromeRouter).isPair(asset) == false) {
```
Comment (RedVeil) [inline src/vault/strategy/compounder/velodrome/VelodromeLPCompounder.sol:17]:
```suggestion
        if (!IVelodromeRouter(velodromeRouter).isPair(asset)) {
```

File: src/vault/strategy/compounder/velodrome/VelodromeLPCompounder.sol [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+        address asset,
+        address router,
+        bytes memory optionalData
+    ) internal override {
+        address token = asset;
+
+        if (baseAsset != token)
+            IERC20(baseAsset).approve(router, type(uint256).max);
+
+        IERC20(0x3c8B650257cFb5f272f799F5e2b4e65093a11a05).approve(
```
Comment (RedVeil) [inline src/vault/strategy/compounder/velodrome/VelodromeLPCompounder.sol:44]: Why is this harcoded? Whats that address?

File: src/vault/strategy/compounder/velodrome/VelodromeLPCompounder.sol [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+            router,
+            type(uint256).max
+        );
+
+        IERC20(ILpToken(token).token0()).approve(router, type(uint256).max);
+
+        IERC20(ILpToken(token).token1()).approve(router, type(uint256).max);
+
+        address velodromeRouter = abi.decode(optionalData, (address));
```
Comment (RedVeil) [inline src/vault/strategy/compounder/velodrome/VelodromeLPCompounder.sol:53]: How is the velodromeRouter different from the normal router?

File: src/vault/strategy/compounder/velodrome/VelodromeLPCompounder.sol [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+    function _getAsset(
+        address baseAsset,
+        address asset,
+        address router,
+        bytes memory toAssetPath,
+        bytes memory optionalData
+    ) internal override {
+        ILpToken LpToken = ILpToken(asset);
+
+        address token = asset;
```
Comment (RedVeil) [inline src/vault/strategy/compounder/velodrome/VelodromeLPCompounder.sol:66]:
```suggestion
```

File: src/vault/strategy/compounder/velodrome/VelodromeLPCompounder.sol [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+
```
Comment (RedVeil) [inline src/vault/strategy/compounder/velodrome/VelodromeLPCompounder.sol:67]: We should first 50/50 trade the base asset to tokenA and tokenB here. This allows the base VelodromeCompounder to simply trade for a single asset.

File: src/vault/strategy/compounder/velodrome/VelodromeCompounder.sol (hunk 1 above)
Comment (amatureApe) [reply src/vault/strategy/compounder/velodrome/VelodromeCompounder.sol:182]: I was a bit confused by this. What is the difference between asset and baseAsset in terms of an lpToken? Users would be depositing a normal lpToken or are they depositing one o the tokens in lpToken and we make the pair for them? But yea can move logic into _getAsset

File: src/vault/strategy/compounder/velodrome/VelodromeLPCompounder.sol (hunk 3 above)
Comment (amatureApe) [reply src/vault/strategy/compounder/velodrome/VelodromeLPCompounder.sol:44]: The address is Velo. This was the issue with the init. I'll check it out and see if I can get it to work.

**Pull request merged**